pkg/base1/test-http.js

```diff
@@ -131,6 +131,36 @@ QUnit.test("streaming", assert => {
             });
 });
 
+QUnit.only("split UTF8 frames", assert => {
+    const done = assert.async();
+    assert.expect(1);
+
+    cockpit.http(test_server)
+            .get("/mock/split-utf8")
+            .then(resp => assert.equal(resp, "initialfirst half é second halffinal", "correct response"))
+            .finally(done);
+});
+
+QUnit.only("truncated UTF8 frame", assert => {
+    const done = assert.async();
+    assert.expect(3);
+    let received = "";
+
+    cockpit.http(test_server)
+            .get("/mock/truncated-utf8")
+            .stream(block => { received += block })
+            .then(() => assert.ok(false, "should not have succeeded"))
+            // does not include the first byte of é
+            .catch(ex => {
+                // does not include the first byte of é
+                assert.equal(received, "initialfirst half ", "received expected data");
+                assert.equal(ex.problem, "protocol-error", "error code");
+                // FIXME: the message is also "protocol-error", huh?
+                assert.equal(JSON.stringify(ex), "Received incomplete UTF-8 message before EOF", "error message");
```

Comment:
@allisonkarlitskaya do you have any idea what's wrong here? Of course the `stringify()` is complete nonsense, it's just for debugging. The channel sends it alright:
```
cockpit.protocol-DEBUG: sending control message {'message': 'Received incomplete UTF-8 message before EOF', 'problem': 'protocol-error'} {'command': 'close', 'channel': '1:2'}
```
but the assertion doesn't see it correctly:
```
g: not ok 4 - error message, expected: "Received incomplete UTF-8 message before EOF", got: "{\"problem\":\"protocol-error\",\"message\":\"protocol-error\"}", test: truncated UTF8 frame
```
WTH happens here?

Reply:
Stupid bug in cockpit.js, found it and made a separate commit.